src/Workleap.Extensions.MediatR.Tests/PreAndPostProcessorTests.cs

```diff
@@ -0,0 +1,180 @@
+﻿using GSoft.Extensions.Xunit;
+using MediatR;
+using MediatR.Pipeline;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Workleap.Extensions.MediatR.Tests;
+
+public sealed class PreAndPostProcessorTests : BaseUnitTest<PreAndPostProcessorTests.Fixture>
+{
+    public PreAndPostProcessorTests(Fixture fixture, ITestOutputHelper testOutputHelper)
+        : base(fixture, testOutputHelper)
+    {
+    }
+
+    [Fact]
+    public async Task SendAsync_Executes_Automatically_Registered_Pre_And_Post_Processors()
```

Comment:
Tested without the fix on MediatR:
* 12.0.1: tests are green
* 12.1.0 & 12.2.0: test fails
* 12.2.0 with our hack: tests are green